framework/tests/value-ignore-range-overflow/RefinementGT.java

```diff
@@ -0,0 +1,58 @@
+import org.checkerframework.common.value.qual.IntRange;
+
+public class RefinementGT {
+
+    void test_forward(int a, int j, int s) {
+        /** forwards greater than */
+        //:: error: (assignment.type.incompatible)
+        @IntRange(from = 0) int aa = a;
+        if (a > -1) {
+            /** a is NN now */
+            @IntRange(from = 0) int b = a;
```

Comment:
I've added additional tests to all of these tests that should cover these cases.